spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/TaskLifecycleListener.java

```diff
@@ -60,12 +60,19 @@
  *     is expected to contain a single application context.</li>
  *     <li>{@link ApplicationReadyEvent} - Used to identify the successful end of a task.</li>
  *     <li>{@link ApplicationFailedEvent} - Used to identify the failure of a task.</li>
+ *     <li>{@link SmartLifecycle#stop()} - Used to identify the end of a task,
+ *     if the {@link ApplicationReadyEvent} or {@link ApplicationFailedEvent}
+ *     is not emitted. This can occur if an error occurs while executing a BeforeTask.
+ *     </li>
  * </ul>
  *
- * <b>Note:</b> By default, the context will be closed at the completion of a task (once
- * the task repository has been updated).  This behavior can be configured via the
- * property <code>spring.cloud.task.closecontext.enable</code> (defaults to true).
- * Also if the context did not start the FailedTask and TaskEnd may not have all the dependencies met.
+ * <b>Note:</b> By default, the context will not be closed at the completion of
+ * a task if there are daemon threads still running.  This behavior can be configured via the
```

Comment:
I'm tweaking the wording of this.  Since 99% of the time, there will not be any daemon threads running, I'd like to word this in the reverse "By default, the context will close at the completion of the task unless other non-daemon threads keep it running".  And it's non-daemon threads...daemon threads do not keep a JVM running.